pkg/proto/merge/merge.go

```diff
@@ -58,18 +70,40 @@ var ReplaceMergeFn MergeFunction = func(dst, src protoreflect.Message) {
 	})
 }
 
+// CatchAllVirtualHostCloneFn creates a clone if src is a catchall virtual host.
+var CatchAllVirtualHostCloneFn CloneFunction = func(dst, src protoreflect.Message) protoreflect.ProtoMessage {
+	catchall := false
+	dst.Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
+		if v.String() == "allow_any" {
```

Comment:
I think this means we are converting each field to a string - may be costly?

Can we just get the real `route.VirtualHost` and get the name field?